src/readmap.py

```diff
@@ -28,15 +89,66 @@ def main():
     args = argparser.parse_args()
 
     if args.p:
-        print(f"Preprocess {args.genome}")
+        #print(f"Preprocess {args.genome}")
+        fasta_recs = read_fasta(args.genome)
+        for fa_rec in fasta_recs:
+            ref = fa_rec[1]
+            SA = SuffixArray(ref)
+            try: genome_name = args.genome.name.split('/')[-1]
+            except: genome_name = args.genome.name
+            write_SA('{}'.format(genome_name),'Preprocessed_{}'.format(fa_rec[0]), SA)
+    
+    elif args.u:
```

Comment:
Du får også et problem hvis dit program forventer en option `-u`, for det får det ikke når jeg tester det... spec'en sagde at programmet bliver kaldt med `-p` for preprocessering og `-d` for readmapping.